src/private/rapidxml_formatter.cpp

```diff
@@ -299,36 +305,122 @@ namespace adm {
+        azimuthNode.addAttribute("coordinate", "azimuth");
+        if (position.has<ScreenEdgeLock>()) {
+            auto screenEdgeLock = position.get<ScreenEdgeLock>();
+            azimuthNode.addOptionalAttribute<HorizontalEdge>(&screenEdgeLock,
+                                                             "screenEdgeLock");
+        }
+        azimuthNode.setValue(position.get<Azimuth>());
+        parentNode.addOptionalElement<AzimuthMin>(&position, name, [](XmlNode& node, AzimuthMin const& min){
+            node.addAttribute("coordinate", "azimuth");
+            node.addAttribute("bound", "min");
+            node.setValue(min.get());
+        });
+        parentNode.addOptionalElement<AzimuthMax>(&position, name, [](XmlNode& node, AzimuthMax const& max){
+            node.addAttribute("coordinate", "azimuth");
+            node.addAttribute("bound", "max");
+            node.setValue(max.get());
+        });
+
+
+        auto elevationNode = parentNode.addNode(name);
+        elevationNode.addAttribute("coordinate", "elevation");
+        if (position.has<ScreenEdgeLock>()) {
+            auto screenEdgeLock = position.get<ScreenEdgeLock>();
+            elevationNode.addOptionalAttribute<VerticalEdge>(&screenEdgeLock,
+                                                             "screenEdgeLock");
+        }
+        elevationNode.setValue(position.get<Elevation>());
+
+        parentNode.addOptionalElement<ElevationMin>(&position, name, [](XmlNode& node, ElevationMin const& min){
+            node.addAttribute("coordinate", "elevation");
+            node.addAttribute("bound", "min");
+            node.setValue(min.get());
+        });
+        parentNode.addOptionalElement<ElevationMax>(&position, name, [](XmlNode& node, ElevationMax const& max){
+            node.addAttribute("coordinate", "elevation");
+            node.addAttribute("bound", "max");
+            node.setValue(max.get());
+        });
+
+        parentNode.addOptionalElement<Distance>(&position, name, [](XmlNode& node, Distance const& distance) {
+            node.addAttribute("coordinate", "distance");
+            node.setValue(distance.get());
+        });
+        parentNode.addOptionalElement<DistanceMin>(&position, name, [](XmlNode& node, DistanceMin const& min){
+            node.addAttribute("coordinate", "distance");
+            node.addAttribute("bound", "min");
+            node.setValue(min.get());
+        });
+        parentNode.addOptionalElement<DistanceMax>(&position, name, [](XmlNode& node, DistanceMax const& max){
+            node.addAttribute("coordinate", "distance");
+            node.addAttribute("bound", "max");
+            node.setValue(max.get());
+        });
+    }
+
+    void formatCartesianSpeakerPosition(
+        XmlNode &parentNode, const std::string &name,
+        const CartesianSpeakerPosition &position) {
+
+      auto xNode = parentNode.addNode(name);
+        xNode.addAttribute("coordinate", "X");
```

Comment:
Could wrap these blocks with `{}` if you want this indent?

Reply:
Thanks, just a formatting error I think (Have discovered I was a bit confused about how my IDE applies auto-indents) so will let clang-format sort that one out.